src/Phan/BlockAnalysisVisitor.php

```diff
@@ -343,10 +343,14 @@ public function visitArrayElem(Node $node): Context
     private function handleScalarStmt(Node $node, Context $context, $child_node): void
     {
         if (\is_string($child_node)) {
+            $consumed = false;
             if (\strpos($child_node, '@phan-') !== false) {
                 // Add @phan-var and @phan-suppress annotations in string literals to the local scope
-                $this->analyzeSubstituteVarAssert($this->code_base, $context, $child_node);
-            } else {
+                $this->analyzeSubstituteVarAssert($this->code_base, $context, $child_node); // TODO Move to an internal plugin?
+                $consumed = true;
+            }
+            $consumed = $consumed || ConfigPluginSet::instance()->analyzeStringLiteralStatement($this->code_base, $context, $child_node);
```

Comment:
```suggestion
            $consumed = ConfigPluginSet::instance()->analyzeStringLiteralStatement($this->code_base, $context, $child_node) || $consumed;
```

Potentially useful for a string literal to contain directives both for phan and its plugins. The short-circuiting logic would mean that something containing `'phan-var'` would not get called for plugins.

Reply:
Ah, my bad, it wasn't my intention to rely on short-circuiting.